### PR TITLE
Template update with HPA behavior block

### PR DESCRIPTION
**Pull request by HebertCL**
# TL;DR

This PR includes `behavior` block to HPA template.

## Summary

In order to fine-tune pod scaling and improve the mechanism and behavior under which we want to scale up/down, we need to include a set of policies that can reflect the scenarios that will make the deployment scale in/out accordingly. This PR adds the `behavior` block to HPA template in order to start working on such fine-tune scaling.

## TODO

This is just an initial effort, and will require a lot of back and forth tweaking and tuning in the Argo template manifest.

## Related tasks

[T-17642](https://pokt.height.app/T-17642)

### CODE REVIEW

File: charts/pokt-portal/templates/hpa.yaml [hunk 1]
```diff
@@ -10,6 +10,17 @@ spec:
     apiVersion: apps/v1
     kind: Deployment
     name: {{ include "pokt-portal.fullname" . }}
+  behavior:
```
Comment (WesleyCharlesBlake) [inline charts/pokt-portal/templates/hpa.yaml:13]: we should include a conditional here , so if autoscaling = true (as with our prod release), it will only add the behaviour block if `.Values.autoscaling.behavior` is provided. other wise it should be ommitted from the HPA resource. This chart would break prod when it gets release because it will expect certain values because `.Values.autoscaling.enabled = true`

**Review by WesleyCharlesBlake — CHANGES_REQUESTED** [review]
all good, one comment about making the behvaiour policies conditional based on input values

File: charts/pokt-portal/templates/hpa.yaml (hunk 1 above)
Comment (HebertCL) [reply charts/pokt-portal/templates/hpa.yaml:13]: Roger that.

Comment (HebertCL) [reply charts/pokt-portal/templates/hpa.yaml]: Comments addressed.

Comment (HebertCL) [reply charts/pokt-portal/templates/hpa.yaml]: Comments addressed.

**Review by WesleyCharlesBlake — APPROVED** [review]
LGTM!

**Review by olakunle-pokt — APPROVED** [review]
